blssig/verifier.go

```diff
@@ -0,0 +1,34 @@
+package blssig
+
+import (
+	"github.com/drand/kyber"
+	bls12381 "github.com/drand/kyber-bls12381"
+	"github.com/drand/kyber/pairing"
+	"github.com/drand/kyber/sign/bdn"
+	"github.com/filecoin-project/go-f3/f3"
+	"golang.org/x/xerrors"
+)
+
+type Verifier struct {
+	suite    pairing.Suite
+	keyGroup kyber.Group
+}
+
+func VerifierWithKeyOnG2() Verifier {
+	suite := bls12381.NewBLS12381Suite()
+	return Verifier{
+		suite:    suite,
+		keyGroup: suite.G2(),
+	}
+}
+
+func (v Verifier) Verify(pubKey f3.PubKey, msg, sig []byte) error {
+	// rejecting zero point pubkeys?
```

Comment:
File an issue for this

Reply:
IIRC zero points were safe, but excluded by some implementations due to an abundance of caution (but don't quote me on that) due to similar issues in EdDSA.
Yeah, I will exclude them.